src/components/Header/HeaderParts/NotificationButton.tsx

```diff
@@ -0,0 +1,241 @@
+        setIsRender(true);
+        isOpenAlarm.current = true;
+      }
+      return !prev;
+    });
+  };
+
+  useOutsideClick(wrapperRef, handleClickButton);
+  useEventSource(eventSource, isOpenAlarm);
+
+  const handleAnimationEnd = () => {
+    if (!isOpen) {
+      setIsRender(false);
+      isOpenAlarm.current = false;
+    }
+  };
+
+  const handleCategoryButtonClick = (value: CategoryType) => {
+    setCurrentCategory(value);
+  };
+
+  const handleDeleteButtonClick = () => {
+    if (currentCategory === '전체' || currentCategory === '커뮤니티') {
+      queryClient.setQueryData(['communityAlarm'], (prev: AlarmAPIDataType) => {
+        prev.alarmDtoList.forEach((alarm) => {
+          deleteAlarmMutate(alarm.id);
+        });
+        return { count: 0, alarmDtoList: [] };
+      });
+    }
+
+    if (currentCategory === '전체' || currentCategory === '상품') {
+      setProductAlarm([]);
+      /* 상품 알림 전체 삭제 */
+    }
+    if (currentCategory === '전체' || currentCategory === '이벤트') {
+      setEventAlarm([]);
+      /* 이벤트 알림 전체 삭제 */
+    }
+  };
+
+  const handleAlarmDataToRead = (id: number, type: AlarmType) => {
+    if (type === 'COMMUNITY') {
+      queryClient.setQueryData(['communityAlarm'], (prev: AlarmAPIDataType) => ({
+        ...prev,
+        alarmDtoList: prev.alarmDtoList.map((alarm) => (alarm.id === id ? { ...alarm, isRead: true } : alarm)),
+      }));
+      postAlarmReadMutate(id);
+      return;
+    }
+    if (type === 'PRODUCT_ORDER') {
+      setProductAlarm((prev) => prev.map((alarm) => (alarm.id === id ? { ...alarm, isRead: true } : alarm)));
+      return;
+    }
+    setEventAlarm((prev) => prev.map((alarm) => (alarm.id === id ? { ...alarm, isRead: true } : alarm)));
+  };
+
+  const handleAlarmDataToDelete = (id: number, type: AlarmType) => {
+    if (type === 'COMMUNITY') {
+      queryClient.setQueryData(['communityAlarm'], (prev: AlarmAPIDataType) => ({
+        count: prev.count - 1 ?? 0,
+        alarmDtoList: prev.alarmDtoList.filter((alarm) => alarm.id !== id),
+      }));
+      deleteAlarmMutate(id);
+      return;
+    }
+
+    if (type === 'PRODUCT_ORDER') {
+      setProductAlarm((prev) => prev.filter((alarm) => alarm.id !== id));
+      /* 상품 알림 삭제 */
+      return;
+    }
+    setEventAlarm((prev) => prev.filter((alarm) => alarm.id !== id));
+    /* 이벤트 알림 삭제 */
+  };
+
+  const handleOpenModal = (value: boolean) => {
+    setIsOpenModal(value);
+  };
+
+  const handleClick = (e: MouseEvent<HTMLDivElement>) => {
+    e.stopPropagation();
+  };
+
+  return (
+    <>
+      <div className={cn('button-wrapper')}>
+        <NotificationIcon
+          width={24}
+          height={24}
+          className={cn('icon', { 'bg-black': isBlack })}
+          onClick={handleClickButton}
+        />
+        {totalUnreadCount > 0 && (
+          <div className={cn('alarm-count', totalUnreadCount > 9 && 'count-more-digit')} onClick={handleClickButton}>
+            {totalUnreadCount > 9 ? '9+' : totalUnreadCount}
```

Comment:
제가 받았던 피드백 중에 하나이지만,, 안하셔도 돼요~ 저스트 공유
```suggestion
            const UNREAD_MAX_COUNT = 9;
            
            {totalUnreadCount > UNREAD_MAX_COUNT ? `${MAX_COUNT}+` : totalUnreadCount}
```

좋은 점은 추후에 수정이 쉽다..?

Reply:
장바구니 아이콘도 적용해야겠네요!